src/txn/txn_rollback_to_stable.c

```diff
@@ -213,7 +232,7 @@ __txn_abort_row_replace_with_hs_value(
      * Otherwise remove the key by adding a tombstone.
      */
     if (valid_update_found) {
-        WT_ERR(__wt_update_alloc(session, hs_value, &upd, &size, WT_UPDATE_STANDARD));
+        WT_ERR(__wt_update_alloc(session, &full_value, &upd, &size, WT_UPDATE_STANDARD));
```

Comment:
If there is no `valid_update_found`, all the calculations that are done are wasted. But this can be changed based on how frequent such scenarios are occurring.

Reply:
Yes, there are potentially many wasted calculations. I can rewrite the code (now or later) to walk the history store twice - the first time to find a valid update and, if it's a modify, the nearest full update. The second to do the removes and the conversion calculations.

Reply:
Hmm, I feel there is no need to traverse the history store twice. We can save the `value` if it is a modify and use them only when the found record is a modify, otherwise just free them?

Reply:
Yes, we could also save a copy of all modifies seen since the last full update.